tweak(docs/native): Update Task Native Enums & Params

DonHulieo opened the pull request:
Changed `TaskHeliMission` & `TaskVehicleMission` parameters and enums, as well as updating their descriptions and styling to be in-line with PR #1128. Also fixed some minor typos in `TaskBoatMission` related in the styling of the first two 'Task_Mission' natives.

Added a small description and the driving style flag enum (see [here](https://github.com/scripthookvdotnet/scripthookvdotnet/blob/230c64db471dc821cacbc9b21bd781ea2abc9311/source/scripting_v3/GTA/VehicleAi/Task/VehicleDrivingFlags.cs) and [here](https://github.com/DurtyFree/gta-v-data-dumps/blob/master/drivingStyleFlagValues.json)). This could be used as reference to the description of drivingStyle to the 'Task_Mission' natives.

AvarianKnight commented on TASK/SetDriveTaskDrivingStyle.md:
````diff
@@ -8,6 +8,8 @@ ns: TASK
+Sets the driving style for a ped currently performing a driving task.
+
 ```
 This native is used to set the driving style for specific ped.  
````
You can remove the native docs here since you cover everything they said anyways

DonHulieo replied:
I wanted to add an example before removing, as well keeping the link for anyone researching. After looking at the link however, it made more sense to incorporate the info into the function description.

AvarianKnight commented on TASK/TaskVehicleMission.md:
````diff
@@ -5,21 +5,49 @@ ns: TASK
+
+If you don't want to use a parameter; pass `0` for `vehicleTarget` and other int parameters, and `-1.0f` for the remaining float parameters.
+
 ```
 missionType: https://alloc8or.re/gta5/doc/enums/eVehicleMissionType.txt
 ```
````
You can remove the native doc as you added the enum

AvarianKnight commented on TASK/TaskVehicleMission.md:
````diff
@@ -5,21 +5,49 @@ ns: TASK
 ```c
 // 0x659427E0EF36BCDE 0x20609E56
-void TASK_VEHICLE_MISSION(Ped driver, Vehicle vehicle, Vehicle vehicleTarget, int missionType, float p4, Any p5, float p6, float p7, BOOL DriveAgainstTraffic);
+void TASK_VEHICLE_MISSION(Ped ped, Vehicle vehicle, Vehicle vehicleTarget, int missionType, float speed, cs_type(Any) int drivingStyle, float radius, float straightLineDist, BOOL DriveAgainstTraffic);
````
```suggestion
void TASK_VEHICLE_MISSION(Ped ped, Vehicle vehicle, Vehicle vehicleTarget, int missionType, float speed, int drivingStyle, float radius, float straightLineDist, BOOL DriveAgainstTraffic);
```
You don't need to to `cs_type` for `int` types

AvarianKnight commented on TASK/TaskHeliMission.md:
````diff
@@ -5,9 +5,13 @@ ns: TASK
 ```c
 // 0xDAD029E187A2BEB4 0x0C143E97
-void TASK_HELI_MISSION(Ped pilot, Vehicle aircraft, Vehicle targetVehicle, Ped targetPed, float destinationX, float destinationY, float destinationZ, int missionFlag, float maxSpeed, float landingRadius, float targetHeading, int unk1, int unk2, cs_type(Hash) float unk3, int landingFlags);
+void TASK_HELI_MISSION(Ped ped, Vehicle heli, Vehicle vehicleTarget, Ped pedTarget, float x, float y, float z, int missionType, float speed, float radius, float heading, cs_type(Int) float height, float minHeight, cs_type(Hash) float slowDist, int missionFlags);
````
```suggestion
void TASK_HELI_MISSION(Ped ped, Vehicle heli, Vehicle vehicleTarget, Ped pedTarget, float x, float y, float z, int missionType, float speed, float radius, float heading, cs_type(int) float height, float minHeight, cs_type(Hash) float slowDist, int missionFlags);
```

AvarianKnight commented on TASK/TaskHeliMission.md:
````diff
@@ -5,9 +5,13 @@ ns: TASK
+
+If you don't want to use a parameter; pass `0.0f` for `x`, `y` and `z`, `0` for `pedTarget`, `vehicleTarget` and other int parameters, and `-1.0f` for the remaining float parameters.
+
 ```
````
You can remove the native doc

DonHulieo commented:
This should be all suggested changes (so far) implemented. 
Also updated all references to drivingStyle/driveMode with link back to function with enum.
Added lua examples in functions where I removed native docs as I felt that info was necessary before removing.

AvarianKnight commented on TASK/SetDriveTaskDrivingStyle.md:
````diff
@@ -8,16 +8,75 @@ ns: TASK
+## Examples
 
+```lua
+local model = `adder`
+RequestModel(model)
+repeat Wait(0) until HasModelLoaded(model)
+local ped = PlayerPedId() -- Player needs to be in a vehicle for this to work
+local coords = GetEntityCoords(ped) - GetEntityForwardVector(ped) * 15.0
+local vehicle = CreateVehicle(model, coords.x, coords.y, coords.z, GetEntityHeading(ped), true, false)
+model = `a_m_m_skater_01`
+RequestModel(model)
+repeat Wait(0) until HasModelLoaded(model)
+local driver = CreatePedInsideVehicle(vehicle, 0, model, -1, true, false)
+TaskVehicleChase(driver, ped)
+SetDriveTaskDrivingStyle(driver, 786468)
+-- Driving Style: DrivingModeAvoidVehiclesReckless
+SetPedKeepTask(driver, true)
````
```suggestion
local vehicleModel = `adder`
RequestModel(vehicleModel)
repeat Wait(0) until HasModelLoaded(vehicleModel)

local ped = PlayerPedId() -- Player needs to be in a vehicle for this to work
local coords = GetEntityCoords(ped) - GetEntityForwardVector(ped) * 15.0
local vehicle = CreateVehicle(model, coords.x, coords.y, coords.z, GetEntityHeading(ped), true, false)
local pedModel = `a_m_m_skater_01`

-- allow the game engine to cleanup the vehicle model if it is no longer needed
SetModelAsNoLongerNeeded(model)

RequestModel(pedModel)
repeat Wait(0) until HasModelLoaded(pedModel)

local driver = CreatePedInsideVehicle(vehicle, 0, model, -1, true, false)

-- allow the game engine to cleanup the ped model if it is no longer needed
SetModelAsNoLongerNeeded(pedModel)

TaskVehicleChase(driver, ped)
SetDriveTaskDrivingStyle(driver, 786468)
-- Driving Style: DrivingModeAvoidVehiclesReckless
SetPedKeepTask(driver, true)
```

The code feels *very* cramped, also your example never releases the model when it is done.

DonHulieo replied:
Ah fair enough, was trying to fit as much code in a little area. Thank you for the catch of not releasing the model, was *'released'* from my mind apparently.
Will reflect these suggestions through all my examples.

DonHulieo commented:
I've pushed requested changes to the examples, is there anything else you'd like me to update?

Is it also worth updating `TaskPlaneMission` in the same vein?

AvarianKnight commented on TASK/SetDriveTaskDrivingStyle.md:
```diff
@@ -8,16 +8,81 @@ ns: TASK
+repeat Wait(0) until HasModelLoaded(vehicle_model)
+
+local ped = PlayerPedId() -- Player needs to be in a vehicle for this to work
+local coords = GetEntityCoords(ped) - GetEntityForwardVector(ped) * 15.0
```
Is this the order of operation you intended?

DonHulieo replied:
Yes as the vehicle is chasing the player ped, so I'm spawning that vehicle behind the player ped.

AvarianKnight commented on TASK/TaskBoatMission.md:
````diff
@@ -8,22 +8,14 @@ ns: TASK
-```
-
 ```c
 enum BoatMissionFlags
````
```suggestion
enum eBoatMissionFlags
```

AvarianKnight commented on TASK/SetDriveTaskDrivingStyle.md:
````diff
@@ -8,16 +8,81 @@ ns: TASK
+Each flag in the `VehicleDrivingFlags` enum can be combined to create a driving style, with each enabling or disabling a specific driving behavior. The driving style can be set to one of the predefined driving styles, or a custom driving style can be created by combining the flags. This can be done by using the bitwise OR operator (`|`) to combine the flags or by adding the decimal values of the  ...
+
+```c
+enum VehicleDrivingFlags
````
```suggestion
enum eVehicleDrivingFlags
```

AvarianKnight commented on TASK/TaskVehicleMission.md:
````diff
@@ -5,21 +5,69 @@ ns: TASK
+If you don't want to use a parameter; pass `0` for int parameters, and `-1.0f` for the remaining float parameters.
+
+```c
+enum VehicleMissionType
````
```suggestion
enum eVehicleMissionType
```

AvarianKnight commented on TASK/SetDriveTaskDrivingStyle.md:
```diff
@@ -8,16 +8,81 @@ ns: TASK
-http://gtaforums.com/topic/822314-guide-driving-styles/  
+Sets the driving style for a ped currently performing a driving task.
+
+Each flag in the `VehicleDrivingFlags` enum can be combined to create a driving style, with each enabling or disabling a specific driving behavior. The driving style can be set to one of the predefined driving styles, or a custom driving style can be created by combining the flags. This can be done by using the bitwise OR operator (`|`) to combine the flags or by adding the decimal values of the  ...
```
```suggestion
Each flag in the `eVehicleDrivingFlags` enum can be combined to create a driving style, with each enabling or disabling a specific driving behavior. The driving style can be set to one of the predefined driving styles, or a custom driving style can be created by combining the flags. This can be done by using the bitwise OR operator (`|`) to combine the flags or by adding the decimal values of the flags together.
```

AvarianKnight commented on TASK/SetDriveTaskDrivingStyle.md:
```diff
@@ -8,16 +8,81 @@ ns: TASK
-* **ped**: 
-* **drivingStyle**: 
+* **ped**: The ped to have their driving style set.
+* **drivingStyle**: The driving style (see `VehicleDrivingFlags`).
```
```suggestion
* **drivingStyle**: The driving style (see `eVehicleDrivingFlags`).
```

AvarianKnight commented on TASK/TaskBoatMission.md:
```diff
@@ -47,13 +39,37 @@ enum BoatMissionFlags
+* **speed**: The speed in m/s (default is -1.0f).
+* **drivingStyle**: The driving style (default is 0) (see [SetDriveTaskDrivingStyle](#_0xDACE1BE37D88AF67)).
+* **radius**: The radius of when the task will be completed (default is -1.0f).
+* **missionFlags**: The mission flags (default is 0) (see `BoatMissionFlags`).
```
```suggestion
* **missionFlags**: The mission flags (default is 0) (see `eBoatMissionFlags`).
```

AvarianKnight reviewed (commented):
Minor consistency requests

AvarianKnight commented on TASK/TaskHeliMission.md:
````diff
@@ -5,43 +5,75 @@ ns: TASK
+If you don't want to use a parameter; pass `0.0f` for `x`, `y` and `z`, `0` for `pedTarget`, `vehicleTarget`, `0` for other int parameters, and `-1.0f` for the remaining float parameters.
+
+```c
+enum HeliMissionFlags
````
```suggestion
enum eHeliMissionFlags
```

AvarianKnight commented on TASK/TaskHeliMission.md:
```diff
@@ -5,43 +5,75 @@ ns: TASK
+* **height**: The height the helicopter will fly at (default is -1.0f).
+* **minHeight**: The height the helicopter will not fly below (default is -1.0f).
+* **slowDist**: The distance to the target at which the helicopter will slow down (default is -1.0f).
+* **missionFlags**: The mission flags (default is 0) (see `HeliMissionFlags`).
```
```suggestion
* **missionFlags**: The mission flags (default is 0) (see `eHeliMissionFlags`).
```

AvarianKnight commented on TASK/TaskVehicleMission.md:
```diff
@@ -5,21 +5,69 @@ ns: TASK
+* **ped**: The ped to be tasked.
+* **vehicle**: The vehicles' entity handle.
+* **vehicleTarget**: The target vehicle.
+* **missionType**: The mission type (default is 0) (see `VehicleMissionType`).
```
```suggestion
* **missionType**: The mission type (default is 0) (see `eVehicleMissionType`).
```
I missed one :face_exhaling:
